nodejs/eks/nodegroup.ts

```diff
@@ -626,3 +626,237 @@ function tagsToAsgTags(tagsInput: InputTags): pulumi.Output<string> {
+    * worker nodes. If you specify ec2_ssh_key, but do not specify this
+    * configuration when you create an EKS Node Group, port 22 on the worker
+    * nodes is opened to the Internet (0.0.0.0/0).
+    *
+    * See for more details:
+    * https://docs.aws.amazon.com/eks/latest/userguide/sec-group-reqs.html
+    */
+    sourceSecurityGroupIds?: pulumi.Output<string[]>;
+}
+
+export interface ManagedNodeGroupScalingConfig {
+    /**
+     * The number of worker nodes that should be running in the cluster. Defaults to 2.
+     */
+    desiredCapacity?: pulumi.Input<number>;
+
+    /**
+     * The minimum number of worker nodes running in the cluster. Defaults to 1.
+     */
+    minSize?: pulumi.Input<number>;
+
+    /**
+     * The maximum number of worker nodes running in the cluster. Defaults to 2.
+     */
+    maxSize?: pulumi.Input<number>;
+}
+
+/**
+ * ManagedNodeGroupOptions describes the configuration options accepted by an
+ * EKS Managed NodeGroup.
+ */
+export interface ManagedNodeGroupOptions extends ManagedNodeGroupBaseOptions {
+    /**
+     * The target EKS cluster.
+     */
+    cluster: Cluster | CoreData;
+}
+
+export interface ManagedNodeGroupData {
+    /**
+     * The EKS managed node group.
+     */
+    nodeGroup: aws.eks.NodeGroup;
+}
+
+/**
+ * ManagedNodeGroup is a component that wraps the AWS managed node group of EC2 instances that provide compute capacity for an EKS cluster.
+ */
+export class ManagedNodeGroup extends pulumi.ComponentResource implements ManagedNodeGroupData {
+    /**
+     * The security group for the cluster's nodes.
+     */
+    public readonly nodeGroup: aws.eks.NodeGroup;
+
+    /**
+     * Create a new AWS managed node group
+     *
+     * @param name The _unique_ name of this component.
+     * @param args The arguments for the node group.
+     * @param opts A bag of options that control this component's behavior.
+     */
+    constructor(name: string, args: ManagedNodeGroupOptions, opts?: pulumi.ComponentResourceOptions) {
+        super("eks:index:ManagedNodeGroup", name, args, opts);
+
+        const group = createManagedNodeGroup(name, args, this);
+        this.nodeGroup = group.nodeGroup;
+        this.registerOutputs(undefined);
+    }
+}
+
+export function createManagedNodeGroup(name: string, args: ManagedNodeGroupOptions, parent: pulumi.ComponentResource): ManagedNodeGroupData {
+    const core = isCoreData(args.cluster) ? args.cluster : args.cluster.core;
+
+    if (!args.instanceRole) {
+        throw new Error(`an instanceRole is required for the managed node group`);
+    }
+
+    // Configure the settings for the autoscaling group.
+    let desiredCapacity: pulumi.Input<number> = 0;
+    let minSize: pulumi.Input<number> = 0;
+    let maxSize: pulumi.Input<number> = 0;
+    if (!args.scalingConfig || args.scalingConfig.desiredCapacity === undefined) {
+        desiredCapacity = 2;
+    } else { desiredCapacity = args.scalingConfig.desiredCapacity; }
```

Comment:
just for consistency we can format these as normal with new lines. 

And total nit - this whole thing could be `const desiredCapacity = args.scalingConfig && args.scalingConfig.desiredCapacity || 2`.